### PR TITLE
HDDS-12869. Add some corner cases in create bucket

### PR DESCRIPTION
**Pull request by rajeshkumarchandolu**
## What changes were proposed in this pull request?

This change extends the S3 gateway smoketest coverage for create-bucket behavior aligned with AWS naming rules and related error responses.

**What and why**

HDDS-12869 adds Robot tests for create-bucket corner cases: invalid bucket names (length, character set, uppercase, leading/trailing punctuation, IP-shaped names, adjacent periods, all-numeric names) and validation that duplicate create returns the expected error when the requester already owns the bucket. The S3 gateway already returns the appropriate error codes; this PR adds smoketest coverage so regressions are caught in compose/CI.

**Changes**

In `hadoop-ozone/dist/src/main/smoketest/s3/bucketcreate.robot`, the following test cases are added or updated:

- Name too short (`ab`) → `InvalidBucketName`
- Name too long (64 characters) → `InvalidBucketName`
- All-uppercase bucket name → `InvalidBucketName`
- Mixed-uppercase bucket name → `InvalidBucketName`
- Invalid characters (`test#12-$`) → error message contains `Invalid bucket name` (AWS CLI client validation)
- Trailing hyphen on otherwise valid name → `InvalidBucketName`
- Two adjacent periods → `InvalidBucketName`
- IP-address-shaped name → `InvalidBucketName`
- Leading dash (`--bucket=-test`) → `InvalidBucketName`
- Leading period → `InvalidBucketName`
- Trailing period → `InvalidBucketName`
- All-numeric name → `InvalidBucketName`
- Maximum valid length (63 characters) → success

Duplicate create with the same name and the same credentials is unchanged; it remains covered by the existing `Create bucket which already exists` test, which expects `BucketAlreadyOwnedByYou`.

The 63-character success case uses a unique 63-character name (`b` plus 62 random lowercase characters) so repeated runs on the same compose cluster do not fail with `BucketAlreadyOwnedByYou` on a fixed bucket name.

Pre-existing cases in the same file (`Create new bucket`, default group ACL, web-endpoint-style bucket names, bucket ownership verification) are unchanged and still run as part of the suite.

No S3 gateway or OM code changes are included.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-12869

## How was this patch tested?

Manual smoketest on the docker-compose `ozone` environment:

1. Build dist smoketest artifacts:
   `mvn clean install -DskipTests -f hadoop-ozone/dist/pom.xml`

2. From `hadoop-ozone/dist/target/ozone-*-SNAPSHOT/compose/ozone`, start compose (recreate containers after a dist rebuild if needed):
   `docker compose up -d`

3. Run the robot suite:
   `../test-single.sh s3g s3/bucketcreate.robot`

   Note: `Test buckets named like web endpoints` uses fixed bucket names (`conf`, `jmx`, …). On a cluster where those buckets already exist from a prior run, delete them or recreate the compose environment before re-running the suite.

**Result:** 19 tests, 19 passed, 0 failed.

```
Using Docker Compose v2
==============================================================================
Bucketcreate :: S3 gateway test with aws cli
==============================================================================
Create new bucket                                                     | PASS |
------------------------------------------------------------------------------
Create bucket with maximum valid name length                          | PASS |
------------------------------------------------------------------------------
Create bucket which already exists                                    | PASS |
------------------------------------------------------------------------------
Create bucket with invalid bucket name                                | PASS |
------------------------------------------------------------------------------
Create bucket with name too short                                     | PASS |
------------------------------------------------------------------------------
Create bucket with name too long                                      | PASS |
------------------------------------------------------------------------------
Create bucket with all uppercase characters in bucket name            | PASS |
------------------------------------------------------------------------------
Create bucket with mixed uppercase characters in bucket name          | PASS |
------------------------------------------------------------------------------
Create bucket validate names can consist only of lowercase letters... | PASS |
------------------------------------------------------------------------------
Create bucket validate names must begin and end with a letter or n... | PASS |
------------------------------------------------------------------------------
Create bucket validate names must not contain two adjacent periods.   | PASS |
------------------------------------------------------------------------------
Create bucket validate names must not be formatted as an IP addres... | PASS |
------------------------------------------------------------------------------
Create bucket validate names must not contain leading dash            | PASS |
------------------------------------------------------------------------------
Create bucket validate names must not contain leading period          | PASS |
------------------------------------------------------------------------------
Create bucket validate names must not contain Trailing period         | PASS |
------------------------------------------------------------------------------
Create bucket validate names must not contain all numeric             | PASS |
------------------------------------------------------------------------------
Create new bucket and check default group ACL                         | PASS |
------------------------------------------------------------------------------
Test buckets named like web endpoints                                 | PASS |
------------------------------------------------------------------------------
Check bucket ownership verification                                   | PASS |
------------------------------------------------------------------------------
Bucketcreate :: S3 gateway test with aws cli                          | PASS |
19 tests, 19 passed, 0 failed
==============================================================================
```

No unit tests or product code changes in this PR. No UI changes.

### CODE REVIEW

File: hadoop-ozone/dist/src/main/smoketest/s3/bucketcreate.robot
Comment (sreejasahithi) [inline hadoop-ozone/dist/src/main/smoketest/s3/bucketcreate.robot:1]: We could also add some more cases like names with:
- Leading dash
- Leading period
- Trailing period
- All numeric

**Review by Gargi-jais11 — COMMENTED** [review]
Thanks @rajeshkumarchandolu for adding robot tests.
+1 to @sreejasahithi comment.
Please check this [aws documentation for naming rules](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html#general-purpose-bucket-names) and add one robot test for each rules which are currently supported by ozone. This was we can verify each of the rules are satisfied.

File: hadoop-ozone/dist/src/main/smoketest/s3/bucketcreate.robot
Comment (rajeshkumarchandolu) [reply hadoop-ozone/dist/src/main/smoketest/s3/bucketcreate.robot:1]: @sreejasahithi  can you validate now . i had added all the scenarios mentioned

**Review by Gargi-jais11 — APPROVED** [review]
LGTM!, +1

**Review by sreejasahithi — APPROVED** [review]
Thanks @rajeshkumarchandolu 
LGTM

File: hadoop-ozone/dist/src/main/smoketest/s3/bucketcreate.robot [hunk 1]
```diff
@@ -42,6 +47,59 @@ Create bucket with invalid bucket name
...
+
+Create bucket validate names must not contain leading dash
+    ${result} =         Execute AWSS3APICli and checkrc         create-bucket --bucket=-test   255
+                        Should contain          ${result}           InvalidBucketName
+
+Create bucket validate names must not contain leading period
+    ${result} =         Execute AWSS3APICli and checkrc         create-bucket --bucket .test   255
+                        Should contain          ${result}           InvalidBucketName
+
+Create bucket validate names must not contain Trailing period
```
Comment (ashishkumar50) [inline hadoop-ozone/dist/src/main/smoketest/s3/bucketcreate.robot]: nit: `trailing period`

File: hadoop-ozone/dist/src/main/smoketest/s3/bucketcreate.robot [hunk 2]
```diff
@@ -42,6 +47,59 @@ Create bucket with invalid bucket name
...
+                        Should contain          ${result}           InvalidBucketName
+
+Create bucket with mixed uppercase characters in bucket name
+    ${randStr} =        Generate Random String     8    [LOWER]
+    ${result} =         Execute AWSS3APICli and checkrc         create-bucket --bucket BuCkEt-${randStr}   255
+                        Should contain          ${result}           InvalidBucketName
+
+Create bucket validate names can consist only of lowercase letters, numbers, periods (.), and hyphens (-).
+    ${result} =         Execute AWSS3APICli and checkrc         create-bucket --bucket test#12-$   255
+                        Should contain          ${result}           Invalid bucket name
```
Comment (ashishkumar50) [inline hadoop-ozone/dist/src/main/smoketest/s3/bucketcreate.robot]: `test#12-$ `appears to be rejected by the AWS CLI client before the request reaches the S3 Gateway. Since this PR is intended to provide Ozone S3 Gateway smoketest coverage, this case doesn't provide regression coverage for Ozone. 
It could be addressed separately in another PR.

**Review by ashishkumar50 — COMMENTED** [review]
@rajeshkumarchandolu Thanks for the patch, overall LGTM except few nits.

**Review by ashishkumar50 — APPROVED** [review]
LGTM
